app/src/lib/stores/repositories-store.ts

```diff
@@ -485,4 +510,38 @@ export class RepositoriesStore extends BaseStore {
 
     return record!.lastPruneDate
   }
+
+  public async isBranchProtected(
+    gitHubRepository: GitHubRepository,
+    branchName: string
+  ): Promise<boolean> {
+    if (gitHubRepository.dbID === null) {
+      return fatalError(
+        'unable to get protected branches, GitHub repository has a null dbID'
+      )
+    }
+
+    const key = getKey(gitHubRepository.dbID, branchName)
+
+    const existing = this.branchProtectionCache.get(key)
+    if (existing !== undefined) {
+      return existing
+    }
+
+    const result = await this.db.protectedBranches
+      .where('[repoId+name]')
+      .equals([gitHubRepository.dbID, branchName])
+      .first()
```

Comment:
This could be simplified to a primary key lookup

```ts
const existing = this.db.protectedBranches.get([gitHubRepository.dbID, branchName])
const protected = existing !== undefined
```

Reply:
This doesn't seem to work OOTB because it doesn't seem to like my usage of `dbID` (a number):

```
Argument of type '(string | number)[]' is not assignable to parameter of type '{ [key: string]: IndexableType; }'.
  Index signature is missing in type '(string | number)[]'.
```

I'll see whether that this is some typing issue with Dexie that I need to workaround ala `PullRequestKey`, but if you've got any extra insight here it'd be greatly appreciated:

https://github.com/desktop/desktop/blob/04cb3aaefed772f3d1657f20c87886727fdac998/app/src/lib/databases/pull-request-database.ts#L63-L70

Reply:
Addressed in 424bc9178618e5767956e29dbdf349ededb60743